chainerx_cc/chainerx/backward.cc

```diff
@@ -211,27 +213,34 @@ class BackwardImpl {
           backprop_id_{backprop_id},
           double_backprop_{double_backprop},
           array_node_grad_map_{std::move(array_node_grad_map)} {
+        // The use of non-POD static storage object here is safe, because destructing a shared_ptr with nullptr does not incur any
+        // destruction order problem.
+        static const std::shared_ptr<ArrayNode> kNullArrayNode{nullptr};
+
         for (const Array& output : outputs) {
-            if (!output.IsBackpropRequired(backprop_id)) {
-                throw ChainerxError{"Cannot start backprop from an array whose gradient is not required (on graph '", backprop_id, "')"};
+            if (output.IsBackpropRequired(backprop_id)) {
+                output_array_nodes_.emplace_back(internal::GetArrayBody(output)->GetArrayNode(backprop_id));
+            } else {
+                output_array_nodes_.emplace_back(kNullArrayNode);
```

Comment:
You can just write as follows (and avoid the static shared_ptr)?
```cpp
output_array_nodes_.reserve(outputs.size());
for (const Array& output : outputs) {
    output_array_nodes_.emplace_back(internal::GetArrayBody(output)->GetArrayNode(backprop_id));
}
```

Reply:
Fixed as suggested.